src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java

```diff
@@ -31,10 +31,17 @@
   public String convert(String input) throws OptionsParsingException {
     if (input.isEmpty()) {
       // TODO(philwo) - replace these deprecated names with more logical ones (e.g. k8 becomes
-      // linux-x86_64, darwin includes the CPU architecture, ...).
+      // linux-x86_64 ...).
       switch (OS.getCurrent()) {
         case DARWIN:
-          return "darwin";
+          switch (CPU.getCurrent()) {
+            case X86_64:
+              return "darwin_x86_64";
```

Comment:
Changing "darwin" to "darwin_x86_64" might cause some issues. @katre is probably more familiar with this.

Reply:
It indeed broke the `test_disable_cc_toolchain_detection` test on macOS, check the latest commit for the fix. I'm not sure if there are more issues I missed.

Reply:
Long term this will be great because there's existing ambiguity between darwin and darwin_x86_64